### ISSUE TITLE
V2 using recursion in source code

### ISSUE THREAD

**Issue by dmitrykurmanov** [hyperapp/hyperapp#730]
Hello! Is it ok to use recursion calls in hyperapp? 

For example in `dispatch` function. 

Could you please check this article http://2ality.com/2014/04/call-stack-size.html
and SO question https://stackoverflow.com/questions/37224520/are-functions-in-javascript-tail-call-optimized

**Comment by jorgebucaran** [comment]
@dmitrykurmanov Yes. Even if the max stack size was a 100, it doesn't really make much sense for an action to return a function that returns a function that... 100 times. There's simply no logical use case for that.

**Comment by jorgebucaran** [comment]
@dmitrykurmanov Not at all. It's a perfectly reasonable question and it actually took me a few minutes to write that answer as I hadn't actually thought about it yet.

Thank you! 🙌
